include/myst/thread.h

```diff
@@ -398,22 +431,22 @@ MYST_INLINE int myst_thread_queue_remove_thread(
     myst_thread_t* t = NULL;
     myst_thread_t* prev = NULL;
 
-    for (t = queue->front; t; prev = t, pos++, t = t->qnext)
+    for (t = queue->_front; t; prev = t, pos++, t = t->_qnext)
     {
         if (t == thread)
         {
             found = true;
             if (prev != NULL)
             {
-                prev->qnext = t->qnext;
-                if (t->qnext == NULL)
-                    queue->back = prev;
+                prev->_qnext = t->_qnext;
+                if (t->_qnext == NULL)
+                    queue->_back = prev;
             }
             else
             {
-                queue->front = queue->front->qnext;
-                if (queue->front == NULL)
-                    queue->back = NULL;
+                queue->_front = queue->_front->_qnext;
```

Comment:
The prefixing of fields with underscores seems unrelated to this PR and I would ask that it be backed out and discussed separately. I understand the intent is to enforce data encapsulation in the hopes of discouraging use of various fields in certain contexts, but it would be a reversal of the previous authors design decision and I think the discussion could be handled separately.